src/task/methods/third_place/script.py

```diff
@@ -0,0 +1,750 @@
+    params_model_1a,
+    params_model_2a,
+    params_model_2b,
+    params_model_2c,
+    params_model_3a,
+    params_model_3b,
+    params_model_4a,
+    params_model_4b,
+    params_model_5a,
+    params_model_5b,
+    params_model_5c,
+    params_model_6a,
+    params_model_6b,
+    params_model_6c,
+    params_model_7a,
+    params_model_7b,
+    params_model_8a,
+    params_model_8b,
+    params_model_8c,
+    params_model_8d,
+]
+
+weights = {
+    "w_0": 0.039503611057797205,
+    "w_1": 0.22778329445024798,
+    "w_2": 0.6470451053292054,
+    "w_3": 0.14998376919294348,
+    "w_4": 0.2417986187654036,
+    "w_5": 0.018231625997453538,
+    "w_6": 0.0670756518443389,
+    "w_7": 0.45476025839087764,
+    "w_8": 0.8953651930541969,
+    "w_9": 0.2341721081339142,
+    "w_10": 0.3780523499542964,
+    "w_11": 0.3428728268932388,
+    "w_12": 0.58025233963141,
+    "w_13": 0.815817039069762,
+    "w_14": 0.4865022144120589,
+    "w_15": 0.01707600530474928,
+    "w_16": 0.26623418232070073,
+    "w_17": 0.20727099624448486,
+    "w_18": 0.47400340366957744,
+    "w_19": 0.6758531294442794,
+}
+
+weights = list(weights.values())
+
+
+def fit_and_predict_embedding_nn(x, y, test_x, model_constructor, best_params):
+    model_params, training_params = split_params_to_training_model(best_params)
+    n_dim = model_params["n_dim"]
+    d = TruncatedSVD(n_dim)
+    y = d.fit_transform(y)
+    model = model_constructor(**model_params)
+    model.fit(
+        x,
+        y,
+        epochs=training_params["epochs"],
+        batch_size=training_params["bs"],
+        verbose=0,
+        shuffle=True,
+    )
+    return d.inverse_transform(model.predict(test_x, batch_size=1))
+
+
+print("Reading input files", flush=True)
+train_df = pd.read_parquet(par["de_train"])
+train_df = train_df.sample(frac=1.0, random_state=42)
+columns = ["cell_type", "sm_name"] + train_df.loc[:, "A1BG":].columns.tolist()
+train_df = train_df.loc[:, columns]
+gene_names = [
+    col
+    for col in train_df.columns
+    if col
+    not in {
+        "cell_type",
+        "sm_name",
+        "sm_lincs_id",
+        "SMILES",
+        "split",
+        "control",
+        "index",
+    }
+]
+
+pseudolabel = pd.read_csv("submission(14).csv")
+test_df = pd.read_csv(par["id_map"])
+pseudolabel = pd.concat(
+    [test_df[["cell_type", "sm_name"]], pseudolabel.loc[:, "A1BG":]], axis=1
+)
+
+train_df = pd.concat([train_df, pseudolabel]).reset_index(drop=True)
+v = train_df.loc[:, "A1BG":]
+
+original_x = train_df[["cell_type", "sm_name"]].values
+original_y = train_df.loc[:, "A1BG":].values
+le = LabelEncoder()
+le.fit(original_x.flat)
+new_names = le.transform(original_x.flat).reshape(-1, 2)
+
+x_test = le.transform(test_df[["cell_type", "sm_name"]].values.flat).reshape(-1, 2)
+
+preds = []
+for model, param in zip(models, params):
+    temp_pred = [
+        fit_and_predict_embedding_nn(new_names, original_y, x_test, model, param)
+        for i in range(reps)
+    ]
```

Comment:
This code throws an error for me:

    Reading input files
    Traceback (most recent call last):
      File "/tmp/viash-run-third_place-hMmNEs.py", line 748, in <module>
        temp_pred = [
                    ^
      File "/tmp/viash-run-third_place-hMmNEs.py", line 749, in <listcomp>
        fit_and_predict_embedding_nn(new_names, original_y, x_test, model, param)
      File "/tmp/viash-run-third_place-hMmNEs.py", line 696, in fit_and_predict_embedding_nn
        y = d.fit_transform(y)
            ^^^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.11/site-packages/sklearn/utils/_set_output.py", line 157, in wrapped
        data_to_wrap = f(self, X, *args, **kwargs)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.11/site-packages/sklearn/base.py", line 1152, in wrapper
        return fit_method(estimator, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.11/site-packages/sklearn/decomposition/_truncated_svd.py", line 229, in fit_transform
        X = self._validate_data(X, accept_sparse=["csr", "csc"], ensure_min_features=2)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.11/site-packages/sklearn/base.py", line 605, in _validate_data
        out = check_array(X, input_name="X", **check_params)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.11/site-packages/sklearn/utils/validation.py", line 957, in check_array
        _assert_all_finite(
      File "/usr/local/lib/python3.11/site-packages/sklearn/utils/validation.py", line 122, in _assert_all_finite
        _assert_all_finite_element_wise(
      File "/usr/local/lib/python3.11/site-packages/sklearn/utils/validation.py", line 171, in _assert_all_finite_element_wise
        raise ValueError(msg_err)
    ValueError: Input X contains NaN.

Do you get the same error?

Reply:
yes, it's super odd. If I run the file as script outside of viash I don't get the error

Reply:
It took me too long to figue out :)

When you run the component with `resources/neurips-2023-kaggle/de_train.parquet`, everything works -- which is what you're doing.

However, when running `viash test`, the component is also run with `resources/neurips-2023-data/de_train.parquet`, which has more genes than the `neurips-2023-kaggle` does.

Since the current implementation relies on `path: submission(14).csv`, this causes errors. 

According to the discussion at https://www.kaggle.com/competitions/open-problems-single-cell-perturbations/discussion/458750, `submission(14).csv` is created by version 264 of the notebook they provide. Do you think so, too?